### ISSUE TITLE
Bug: MD072 (frontmatter-key-sort) issues with enabling, LSP, and indentation

### ISSUE THREAD

**Issue by chrisgrieser** [rvben/rumdl#264]
I found multiple minor issues with MD072:

1. with the config:

```toml
[frontmatter-key-sort]
enabled = true
```

`rumdl fmt` reports: `[config warning] Unknown option for rule MD072: enabled`. However, it's a false report, since MD072 is an opt-in rule, and since `enabled` does take effect.

2. MD072 does not work when used with LSP-formatting. Running `vim.lsp.buf.format()` (the formatting command in nvim) does nothing for this rule, while all other rules are correctly formatted.

3. Given this frontmatter:

```yaml
---
bbb: 123
aaa:
  - hello
  - world
---
```

running `rumdl fmt` results in this:

```yaml
---
aaa:
- hello
- world
bbb: 123
---
```

The sorting works correctly, but the indentation of the list is removed.

**Comment by chrisgrieser** [comment]
thank you for the as always quick fix!
